src/logic/safe/store/reducer/gatewayTransactions.ts

```diff
@@ -42,6 +43,34 @@ export type TransactionDetailsPayload = {
 
 type Payload = HistoryPayload | QueuedPayload | TransactionDetailsPayload
 
+const getUpdatedTx = (storedTx: Transaction, newTx: Transaction) => {
+  const isUpdatedTx =
+    isMultisigExecutionInfo(storedTx.executionInfo) &&
+    isMultisigExecutionInfo(newTx.executionInfo) &&
+    storedTx.executionInfo.confirmationsSubmitted !== newTx.executionInfo.confirmationsSubmitted
+
+  return isUpdatedTx
+    ? // Will remove txDetails as they will have changed because of confirmations
+      newTx
+    : // Create new object, preserving txDetails
+      merge({}, storedTx, newTx)
+}
+
+const updateTxs = (txs: Transaction[], newTx: Transaction) => {
+  const newTxs = cloneDeep(txs)
```

Comment:
This can be replaced with `const newTxs = txs.slice()` (shallow copy). Because you're only modifying the array and not the array elements.